worker/worker.go

```diff
@@ -62,6 +82,52 @@ func HandleJoinResponse(response *utils.HttpResponse, w *Worker) {
 	w.QueueId = queueId
 }
 
+func (w *Worker) CheckAuthentication() error {
+	err := utils.CheckToken(w.Token)
+	if err != nil{
+		return errors.New("The worker is not properly authenticated, " +
+			"the token is invalid: " + err.Error())
+	}
+
+	if w.QueueId == "" {
```

Comment:
it seems we are mixing concepts here. Why the Authenticaton check verifies the QueueId?

Reply:
makes sense. I'll change it.